Added accessibility labels to the headers and footers in the theme

andrewpking opened the pull request:
This change adds labels to content that might want to be skimmed by a screen reader. 

### In the header

- The nav bar now has a nav tag.
- The search field is now labeled as a search object
- The search button is now labeled as a button

These fixes will benefit screen reader users by allowing them to semantically navigate the header and get feedback about what element they are interacting with.

### In the footer

- The link to the support page with the text "more..." is now labeled "Support Us", allowing screen readers to understand where this link goes.
- The font sizes have been increased to allow for larger touch targets. This supports people with mobility issues to be able to click on the intended link.

### CSS changes

- Reflowed my styles to be more specific
- Put styles in appropriate places in alphabetical order
- Added comments to my styles to make them easy to find.

Overall, these changes are a good start at making the website more accessible.

andrewpking committed:
BC Logo no longer overlaps text on small screens.

andrewpking committed:
Styles created for donate buttons to be created on production site.

andrewpking committed:
Fix nav issues in header by adding nav tags to DOM.

andrewpking committed:
Added an aria label to search.

andrewpking committed:
Alt text for KBCS logo indicates that it links to homepage.

andrewpking committed:
Now using roles instead of ARIA labels for search.

andrewpking committed:
Search form now includes labels.

andrewpking committed:
Link to support us is more verbose.

andrewpking committed:
Added styles that make the touch targets larger.

andrewpking committed:
Made link for support shorter in footer.

andrewpking committed:
Added subclass to portion of footer with bclogo

andrewpking committed:
Fixed CSS issues caused by larger fonts in footer.

andrewpking committed:
Reflowed styles to be alphabetized in tablet media screen.

taija commented on header.php:
```diff
@@ -162,9 +162,9 @@
                                		 <form id="search" method="get" action="<?php echo esc_url( home_url( '/' ) ); ?>"> 
                                         <span aria-hidden="true" data-icon="&#xf002;"></span>
-                                        <input class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
+                                        <input role="search" class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
```
This is incorrect use of this ARIA landmark. The role of `search` should be applied on the `<form>` tag, not on the input field. If you would like to note that the input is a search input, you can use `type="search"` instead of the current `type="text"`

andrewpking replied:
Resolved in a commit to this branch.

taija commented on header.php:
```diff
@@ -162,9 +162,9 @@
+                                        <input role="search" class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
 										<input type='hidden' name='post_type' value='programs,segments,staff,events,ads' />
-                                         <input id="searchsubmit" value="Search" type="submit" class="btn" />
+                                         <input role="button" id="searchsubmit" value="Search" type="submit" class="btn" />
```
This is an unnecessary, as input with a type of `submit` already has the correct roles natively, as this is a native HTML element.  W3C recommends against using excess ARIA in cases where it is not absolutely necessary, as it can conflict with native roles.

andrewpking replied:
resolved issue in a commit.

taija commented on searchform.php:
```diff
@@ -1,8 +1,8 @@
   <div class="input-append">
 		<span aria-hidden="true" data-icon="&#xf002;"></span>
-		<input class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
+		<input label="Search" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
```
Label is not a valid attribute for `<input>` elements. To make this valid, you would need to add an `id` to the `<input>`, then add a separate `<label>` element with the appropriate text, with a `for` attribute referencing the ID of the input. 

Example: 
```html
<label for="searchquery">Search</label>
<input id="searchquery" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
```

andrewpking replied:
Removed improper labels from this in a commit.

taija commented on searchform.php:
```diff
@@ -1,8 +1,8 @@
+		<input label="Search" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
 		<input type='hidden' name='post_type' value='programs,segments,staff,events,ads' />
-		<button id="searchsubmit" value="Search" type="submit" class="btn">Search</button>
+		<button label="button" id="searchsubmit" value="Search" type="submit" class="btn">Search</button>
```
Label is not a valid attribute for the button element. Not sure what is being attempted here.

andrewpking replied:
Removed this label.

taija reviewed (changes requested):
Please review the noted inline issues.

andrewpking committed:
Update header.php

### Form now has role=search

- Removed improper use of ARIA-Role from input element.
- Moved ARIA-role to form element.

andrewpking committed:
Update searchform.php

### Removed improper use of ARIA-labels

- Removed use of label for input and button
- Will add labels properly in the future when I can test them

andrewpking reviewed (commented):
I have made additional commits to this that resolve all comments. Please review these when you can @taija

taija commented on header.php:
```diff
@@ -126,7 +126,7 @@
 							wp_nav_menu( array( 
 								'menu' => 'main-nav', 
-								'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+								'items_wrap'      => '<nav><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
This is still unresolved- navs must be labelled correctly

andrewpking replied:
Thanks for finding this, I resolved this issue.

taija commented on header.php:
```diff
@@ -181,7 +181,7 @@
 												wp_nav_menu( array( 
 													'menu' => 'main-nav', 
-													'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+													'items_wrap'      => '<nav><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
See above comment regarding navs

andrewpking replied:
Thanks for finding this, I have resolved this issue in a commit.

taija reviewed (changes requested):
See comments

andrewpking committed:
Update header.php

### Added aria-labelledby as requested

- nav elements now include aria-labelledby="main-nav"
- Both nav wrappers use the same label because one is for the mobile app and one is for web.

andrewpking reviewed (commented):
### Resolved unlabeled nav's in header.php

- Nav elements include aria-labelledby="main-nav"
- They use the same label because one is for mobile and the other is for web.
